python/ray/data/dataset.py

```diff
@@ -3586,7 +3586,7 @@ def __repr__(self) -> str:
             schema_str = ", ".join(schema_str)
             schema_str = "{" + schema_str + "}"
         count = self._meta_count()
-        return "Dataset(num_blocks={}, num_rows={}, schema={})".format(
+        return "<Dataset num_blocks={} num_rows={} schema={}>".format(
```

Comment:
cc @clarkzinzow any thoughts

Reply:
actually, @bveeramani  what's your rule here?

Reply:
The `repr`docstring says:

> repr(object) -> string
>
> Return the canonical string representation of the object.
> For most object types, eval(repr(object)) == object.


DeNero says [in the textbook](https://composingprograms.com/pages/27-object-abstraction.html#string-conversion)

> In cases where no representation exists that evaluates to the original value, Python typically produces a description  surrounded by angled brackets.